### PR TITLE
the schema which added more configs.

### PR DESCRIPTION
**Pull request by andyliuliming**
added the 
1. iaas: azure
2. storage:
      os_disk:
      data_disks:
      azure_files:
3. network
4. roles.
5. migrate_from in the database.

### CODE REVIEW

File: spec/spec.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    sku: Standard
+
+services:
+  storage_account:
+    name: storage_account_1
+    location: eastus
+    sku: standard
+  # this is only for the service based database, if you want to deploy the iaas version.
+  # please use the roles instead.
+  database:
```
Comment (gossion) [inline spec/spec.yml]: please make it a list

Comment (gossion) [reply spec/spec.yml]: also for `storage_account`

Comment (andyliuliming) [reply spec/spec.yml]: nice. made the changes :)

File: spec/spec.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      os_disk:
+        size: 10G
+      data_disks:
+        - size: 100G
+        # - name_convension: uuid # this canbe uuid/group, if uuid, one uuid would be gnerated as name, if 
+        #   size: 100G
+      azure_files:
+        - storage_account: storage_account_1
+          name: azure_file_1
+          mount_point: /mnt/azurefile_1
```
Comment (norshtein) [inline spec/spec.yml:27]: What's the function of `azure_files`?

**Review by gossion — APPROVED** [review]

File: spec/spec.yml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  - name: network1
+    vnet: vnet-1
+    subnets:
+      - name: 
+        range: 10.10.0.0/24
+        gateway: 10.10.0.1
+
+load_balancer:
+  - name: load_balancer_1
+    sku: Standard
```
Comment (norshtein) [inline spec/spec.yml:72]: Do we really need a **list** of load balancers in LAMP infra? If so, change the key from `load_balancer` to `load_balancers`

**Comment by norshtein** [comment]
One question: maintaining all things in a single manifest v.s. separating different function units into several manifests, I prefer the second one. Is there any reason to choose the first one?

**Comment by andyliuliming** [comment]
> One question: maintaining all things in a single manifest v.s. separating different function units into several manifests, I prefer the second one. Is there any reason to choose the first one?

let me explain it offline with you :)

File: spec/spec.yml [hunk 4]
```diff
@@ -0,0 +1,96 @@
+---
+schema: v0.1
```
Comment (bingosummer) [inline spec/spec.yml:2]: This `spec.yml` is more like an example.
Shall we use a formal specification?

Comment (andyliuliming) [reply spec/spec.yml:2]: yes, we should investigate use one more formal spec. let's do it later.
filed one issue to track:
https://github.com/kun-lun/artifacts/issues/4

**Pull request merged**